applications/luci-app-adblock/luasrc/view/adblock/runtime.htm

```diff
@@ -1,10 +1,192 @@
 This is free software, licensed under the Apache License, Version 2.0
+local sys     = require("luci.sys")
+
 -%>
+<style type="text/css">
+.runtime
+{
+	color:#0069d6;
+	font-weight:bold;
+	display:inline-block;
+	width:100%;
+	padding-top: 0.5rem;
+}
+</style>
+
+<script type="text/javascript">
+//<![CDATA[
+	function status_update(json)
+	{
+			var view  = document.getElementById("value_1");
+			var btn1  = document.getElementById("btn1");
+			var btn2  = document.getElementById("btn2");
+			var input = json.data.adblock_status;
+
+			view.innerHTML = input || "-";
+			if (input === "enabled")
+			{
+				btn1.value = "Suspend";
+				btn2.value = "Refresh";
+				btn1.disabled = false;
+				running(btn1_running, 0);
+				btn2.disabled = false;
+				running(btn2_running, 0);
+			}
+			else if (input === "paused")
+			{
+				btn1.value = "Resume";
+				btn2.value = "Refresh";
+				btn1.disabled = false;
+				running(btn1_running, 0);
+				btn2.disabled = false;
+				running(btn2_running, 0);
+			}
+			else
+			{
+				btn1.value = "Suspend";
+				btn2.value = "Refresh";
+				btn1.disabled = true;
+				btn2.disabled = true;
+			}
+			view = document.getElementById("value_2");
+			input = json.data.adblock_version;
+			view.innerHTML = input || "-";
+			view = document.getElementById("value_3");
+			input = json.data.fetch_utility;
+			view.innerHTML = input || "-";
+			view = document.getElementById("value_4");
+			input = json.data.dns_backend;
+			view.innerHTML = input || "-";
+			view = document.getElementById("value_5");
+			input = json.data.overall_domains;
+			view.innerHTML = input || "-";
+			view = document.getElementById("value_6");
+			input = json.data.last_rundate;
+			view.innerHTML = input || "-";
+	}
+
+	function btn_action(action)
+	{
+		var btn1 = document.getElementById("btn1");
+		var btn1_running = document.getElementById("btn1_running");
+		var btn2 = document.getElementById("btn2");
+		var btn2_running = document.getElementById("btn2_running");
+
+		btn1.disabled = true;
+		btn2.disabled = true;
+
+		if (action.value === "Refresh")
+		{
+			running(btn2_running, 1);
+		}
+		else
+		{
+			running(btn1_running, 1);
+		}
+
+		new XHR.get('<%=luci.dispatcher.build_url("admin", "services", "adblock")%>/action/' + action.value, null,
+		function(x)
+		{
+			if (!x)
+			{
+				return;
+			}
+		});
+	}
+
+	function running(element, state)
+	{
+		if (state === 1)
+		{
+			var running_html = '<img src="<%=resource%>/icons/loading.gif" alt="<%:Loading%>" width="16" height="16" style="vertical-align:middle" />';
+			element.innerHTML = running_html;
+		}
+		else
+		{
+			element.innerHTML = '';
+		}
+	}
 
-<%+cbi/valueheader%>
+	XHR.get('<%=luci.dispatcher.build_url("admin", "services", "adblock", "status")%>', null,
+	function(x, json_info)
+	{
+		if (!x || !json_info)
+		{
+			var btn1 = document.getElementById("btn1");
+			var btn2 = document.getElementById("btn2");
+			btn1.value = "Suspend";
```

Comment:
Note that you'll lose translations here. Should it be `btn1.value = "<%:Suspend%>";` maybe? I noticed that you use the label strings on the server side to decide which action was invoked, which would fail if the buttons get translated. But you could fix that by using a hidden field or by maybe changing the `name` attribute instead so that you check for `if http.formvalue("do_suspend") then` instead of `if http.formvalue("fixed_name") == "Suspend" then`